packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -0,0 +1,289 @@
+        context.report({
+          node,
+          messageId: 'noMapSpreadInObject',
+        });
+
+        return;
+      }
+
+      if (isArray(checker, type)) {
+        context.report({
+          node,
+          messageId: 'noArraySpreadInObject',
+        });
+
+        return;
+      }
+
+      if (
+        !options.allowIterables &&
+        isIterable(type, checker) &&
+        !isString(type) &&
+        !isMap(services.program, type)
+      ) {
+        context.report({
+          node,
+          messageId: 'noIterableSpreadInObject',
+        });
+
+        return;
+      }
+
+      if (!options.allowClassInstances && isClassInstance(type)) {
+        context.report({
+          node,
+          messageId: 'noClassInstanceSpreadInObject',
+        });
+
+        return;
+      }
+
+      if (
+        !options.allowClassDeclarations &&
+        isClassDeclaration(type) &&
+        !isClassInstance(type)
+      ) {
+        context.report({
+          node,
+          messageId: 'noClassDeclarationSpreadInObject',
+        });
+
+        return;
+      }
+    }
+
+    return {
+      'ArrayExpression > SpreadElement': checkArraySpread,
+      'ObjectExpression > SpreadElement': checkObjectSpread,
+    };
+  },
+});
+
+function isIterable(type: ts.Type, checker: ts.TypeChecker): boolean {
+  return tsutils
+    .typeParts(type)
+    .some(
+      t => !!tsutils.getWellKnownSymbolPropertyOfType(t, 'iterator', checker),
+    );
+}
+
+function isArray(checker: ts.TypeChecker, type: ts.Type): boolean {
+  return isTypeRecurser(
+    type,
+    t => checker.isArrayType(t) || checker.isTupleType(t),
+  );
+}
+
+function isString(type: ts.Type): boolean {
+  return isTypeRecurser(type, t => isTypeFlagSet(t, ts.TypeFlags.StringLike));
+}
+
+function isFunctionWithoutProps(type: ts.Type): boolean {
+  return isTypeRecurser(
+    type,
+    t => t.getCallSignatures().length > 0 && t.getProperties().length === 0,
+  );
+}
+
+function isPromise(program: ts.Program, type: ts.Type): boolean {
+  return isTypeRecurser(type, t => isPromiseLike(program, t));
+}
+
+function isClassInstance(type: ts.Type): boolean {
+  return isTypeRecurser(type, t => {
+    return (
+      t.isClassOrInterface() &&
+      tsutils.isSymbolFlagSet(t.symbol, ts.SymbolFlags.Value)
+    );
+  });
+}
+
+function isClassDeclaration(type: ts.Type): boolean {
+  return isTypeRecurser(type, t => {
+    if (
+      tsutils.isObjectType(t) &&
+      tsutils.isObjectFlagSet(t, ts.ObjectFlags.InstantiationExpressionType)
+    ) {
+      return true;
+    }
+
+    const kind = t.getSymbol()?.valueDeclaration?.kind;
+
+    return (
+      kind === ts.SyntaxKind.ClassDeclaration ||
+      kind === ts.SyntaxKind.ClassExpression
+    );
+  });
+}
+
+function isMap(program: ts.Program, type: ts.Type): boolean {
+  return isTypeRecurser(
+    type,
+    t =>
+      isBuiltinSymbolLike(program, t, 'Map') ||
+      isBuiltinSymbolLike(program, t, 'ReadonlyMap'),
+  );
+}
```

Comment:
Shouldn't it catch `WeakMap` as well?

This case is not reported

```ts
declare const map: WeakMap<{foo: string}, number>;
const mapSpread = { ...map };
```

[playground](https://deploy-preview-8509--typescript-eslint.netlify.app/play/#ts=5.4.5&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gWygBwC54B1EKAawFl8AeAbwDMkkjMYBLFAcwF8AaeCgCu2AEYgYAPgDcAKGTosuPAGU8cKMHgBeeHXgA6I8vg9pQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1kxaIgCGQ9FETRoHaJHBgAviDVA&tsconfig=&tokens=false)

Reply:
So after @Josh-Cena's [comment](https://github.com/typescript-eslint/typescript-eslint/pull/8509#issuecomment-2289631409), I think (?) that now I understand why this is happening.

Is there any similar check in the codebase that I can take as a reference to solve it?